core/domain/customization_args_util.py

```
@@ -92,11 +92,6 @@ def validate_customization_args_and_values(
             % customization_args)
 
     # Validate and clean up the customization args.
-
-    # Populate missing keys with the default values.
-    customization_args = get_full_customization_args(
-        customization_args, ca_specs_to_validate_against)
-
     # Remove extra keys.
```

Should this be tidied up as well as part of the migration (and we raise an error here if there are extra keys)?

---

Should we raise an error if there are extra keys? Or simply remove the extra keys? Also sorry- what do you mean by tidying up?

---

Raise an error if there are extra keys. (Don't worry about the "tidying up" bit -- the previous sentence is what I meant. I.e. we should remove as much of the logic as possible for "cleaning up" the customization args if we can have guarantees that some of these invariants can be taken as assumptions going forward.)

---

Done